### PR TITLE
fix: `page.goto` error throwing on 40x/50x responses with an empty body (#9523)

### PR DESCRIPTION
**Pull request by Rafael-Martins**
The observed problem is:  when running Chrome in headful mode, responses with status 404 and 500 that lack a body are treated as errors.
Solution: bypass the error using the error message.

**What kind of change does this PR introduce?**
bugfix

**Did you add tests for your changes?**
yes

**Summary**

Remove the error throw on the `navigate` method, in order to get [#9523](https://github.com/puppeteer/puppeteer/issues/9523) fixed

### CODE REVIEW

**Comment by Rafael-Martins** [comment]
Hey guys, I'm new here, please tell me if this change makes sense!

**Review by OrKoN — CHANGES_REQUESTED** [review]
I think we cannot just remove the error message here. Did you find out why the browser handles the errors differently? perhaps it should be fixed upstream first.

**Comment by Rafael-Martins** [comment]
> I think we cannot just remove the error message here. Did you find out why the browser handles the errors differently? perhaps it should be fixed upstream first.

Oh, I see, just saw some tests failing because of this as well, I'm digging into it!

**Comment by Rafael-Martins** [comment]
The issue only happen in chrome, looking at the  I see that `errorText` is returned when the navigation is failed, so chrome is considering the `404` as a navigation error. Can we filter it like my last commit?

It appears that the issue at hand is specific to the Chrome browser. By consulting the documentation for the [devtools-protocol docs](https://chromedevtools.github.io/devtools-protocol/tot/Page/#method-navigate) it can be observed that the errorText property is provided when a navigation failure occurs. In this case, Chrome is interpreting a 404 error as a navigation failure. To address this, it may be possible to implement a filter similar to the one utilized in a previous commit, in order to bypass this error.

**Comment by OrKoN** [comment]
@Rafael-Martins so when the browser treats the navigation as an error it shows a different kind of the error page I think which is usually restricted to CDP. What is the situation after the navigation errors, can you still evaluate or navigate the page to a different URL?

**Review by OrKoN — CHANGES_REQUESTED** [review]
Let's add a test for covering this scenario

**Comment by Rafael-Martins** [comment]
> @Rafael-Martins so when the browser treats the navigation as an error it shows a different kind of the error page I think which is usually restricted to CDP. What is the situation after the navigation errors, can you still evaluate or navigate the page to a different URL?

I tested and the navigate works normally, I added 2 tests that covers the `404`,`505` responses with empty body, they look similar or almost redundant, if you think that it is not needed I can remove.

File: test/src/navigation.spec.ts [hunk 1]
```diff
@@ -304,6 +304,30 @@ describe('navigation', function () {
       expect(response.ok()).toBe(false);
       expect(response.status()).toBe(404);
     });
+    it('should not throw a error when get a 404 response with empty body', async () => {
```
Comment (OrKoN) [inline test/src/navigation.spec.ts]:
```suggestion
    it('should not throw an error for a 404 response with an empty body', async () => {
```

File: test/src/navigation.spec.ts [hunk 2]
```diff
@@ -304,6 +304,30 @@ describe('navigation', function () {
...
+      server.setRoute('/404-error', (_, res) => {
+        res.statusCode = 404;
+        res.end();
+      });
+
+      const response = (await page.goto(server.PREFIX + '/404-error'))!;
+      expect(response.ok()).toBe(false);
+      expect(response.status()).toBe(404);
+    });
+    it('should not throw a error when get a 500 response with empty body', async () => {
```
Comment (OrKoN) [inline test/src/navigation.spec.ts]:
```suggestion
    it('should not throw an error for a 500 response with an empty body', async () => {
```

**Review by OrKoN — APPROVED** [review]

**Pull request merged**